Run Email Notification  not working

Users are not getting notifications for email and for meetings
Nothing I could find is simular


#### Issue
Run Email Notification is running but no notification is being sent

#### Expected Behavior
Notifications to be sent

#### Actual Behavior
PHP Notice:  Trying to get property of non-object in /power-choice/web/crm.company.com.au/public_html/crm/modules/Activities/EmailReminder.php on line 163
PHP Fatal error:  Uncaught Error: Call to a member function getReplyToAddress() on null in /power-choice/web/crm.company.com.au/public_html/crm/modules/Activities/EmailReminder.php:163
Stack trace:
#0 /power-choice/web/crm.company.com.au/public_html/crm/modules/Reminders/Reminder.php(212): EmailReminder->sendReminders(Object(Call), Object(Administration), Array)
#1 /power-choice/web/crm.company.com.au/public_html/crm/modules/Activities/EmailReminder.php(97): Reminder::sendEmailReminders(Object(EmailReminder), Object(Administration))
#2 /power-choice/web/crm.company.com.au/public_html/crm/modules/Schedulers/_AddJobsHere.php(435): EmailReminder->process()
#3 /power-choice/web/crm.company.com.au/public_html/crm/modules/SchedulersJobs/SchedulersJob.php(542): sendEmailReminders(Object(SchedulersJob))
#4 /power-choice/web/crm.compnay.com.au/public_html/crm/include/SugarQueue/SugarCronJobs.php(191): SchedulersJob->runJob()
#5 /power-choice/web/crm.companycom.au/public_html/crm/include/SugarQueue/Su in /power-choice/web/crm.company.com.au/public_html/crm/modules/Activities/EmailReminder.php on line 163

#### Possible Fix


#### Steps to Reproduce
run the job get the error.

#### Context
can not run one of the core function of the system

#### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* SuiteCRM Version used:
Version 7.11.2

Sugar Version 6.5.25 (Build 344)
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
![image](https://user-images.githubusercontent.com/2706403/53299880-4ade7c00-3873-11e9-9cfc-f3a0f7ccb524.png)

* Environment name and version (e.g. MySQL, PHP 7):
php 7.1 , mysql 
* Operating System and version (e.g Ubuntu 16.04):
Selinux 7.6

Also getting the following errors 
Mon Feb 25 09:42:01 2019 [62107][1][ERROR] ImapHandler trying to use a non valid resource stream.
Mon Feb 25 09:42:01 2019 [62107][1][ERROR] ImapHandler trying to use a non valid resource stream.
Mon Feb 25 09:42:01 2019 [62107][1][WARN] An Imap error detected: "IMAP search error"
Mon Feb 25 09:42:01 2019 [62107][1][WARN] An Imap error detected: "IMAP expunge error"
Mon Feb 25 09:42:01 2019 [62107][1][WARN] Request ssl value not found.
Mon Feb 25 09:42:01 2019 [62107][1][WARN] Requested folder is not defined
Mon Feb 25 09:42:01 2019 [62107][1][ERROR] ImapHandler trying to use a non valid resource stream.
Mon Feb 25 09:42:01 2019 [62107][1][ERROR] ImapHandler trying to use a non valid resource stream.
Mon Feb 25 09:42:01 2019 [62107][1][WARN] An Imap error detected: "IMAP search error"
Mon Feb 25 09:42:01 2019 [62107][1][WARN] An Imap error detected: "IMAP expunge error"
Mon Feb 25 09:42:01 2019 [62107][1][WARN] An Imap error detected: "SECURITY PROBLEM: insecure server advertised AUTH=PLAIN"
Mon Feb 25 09:42:01 2019 [62107][1][WARN] Request ssl value not found.
Mon Feb 25 09:42:01 2019 [62107][1][WARN] Requested folder is not defined
Mon Feb 25 09:42:01 2019 [62107][1][ERROR] ImapHandler trying to use a non valid resource stream.
Mon Feb 25 09:42:01 2019 [62107][1][ERROR] ImapHandler trying to use a non valid resource stream.
Mon Feb 25 09:42:01 2019 [62107][1][WARN] An Imap error detected: "IMAP search error"
Mon Feb 25 09:42:01 2019 [62107][1][WARN] An Imap error detected: "IMAP expunge error"
Mon Feb 25 09:42:01 2019 [62107][1][WARN] An Imap error detected: "SECURITY PROBLEM: insecure server advertised AUTH=PLAIN"
Mon Feb 25 09:42:01 2019 [62107][1][WARN] Request ssl value not found.

--- a/modules/Activities/EmailReminder.php
+++ b/modules/Activities/EmailReminder.php
@@ -134,25 +134,23 @@ class EmailReminder
 
         if (!empty($bean->created_by)) {
             $user_id = $bean->created_by;
+        } elseif (!empty($bean->assigned_user_id)) {
+            $user_id = $bean->assigned_user_id;
         } else {
-            if (!empty($bean->assigned_user_id)) {
-                $user_id = $bean->assigned_user_id;
-            } else {
-                $user_id = $GLOBALS['current_user']->id;
-            }
+            $user_id = $GLOBALS['current_user']->id;
         }
         $user = BeanFactory::getBean('Users', $user_id);
 
         $OBCharset = $GLOBALS['locale']->getPrecedentPreference('default_email_charset');
-        require_once("include/SugarPHPMailer.php");
+        require_once __DIR__ . '/../../include/SugarPHPMailer.php';
         $mail = new SugarPHPMailer();
         $mail->setMailerForSystem();
 
         if (empty($admin->settings['notify_send_from_assigning_user'])) {
             $from_address = $admin->settings['notify_fromaddress'];
-            $from_name = $admin->settings['notify_fromname'] ? "" : $admin->settings['notify_fromname'];
+            $from_name = $admin->settings['notify_fromname'] ? '' : $admin->settings['notify_fromname'];
         } else {
-            $from_address = $user->emailAddress->getReplyToAddress($user);
+            $from_address = (new SugarEmailAddress())->getReplyToAddress($user);
             $from_name = $user->full_name;
         }
 

